.pre-commit-config.yaml

```diff
@@ -0,0 +1,19 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.3.0
+    hooks:
+    -   id: trailing-whitespace
+    -   id: check-yaml
+    -   id: check-added-large-files
+    -   id: check-executables-have-shebangs
+    -   id: check-merge-conflict
+    -   id: detect-private-key
+    -   id: no-commit-to-branch
+    -   id: check-toml
+-   repo: https://github.com/psf/black
+    rev: 22.6.0
```

Comment:
Wonder if we can just use "latest". Even if it means the occasional surprise, I don't think it will be very frequent, and it would mean we wouldn't have to worry about updating this (in two places) and having all developers keep this version.

Reply:
I would hope that the dependabot would pick this up so it should not be to bad to keep this updated?

Reply:
I think at the moment dependabot is just configured to monitor github actions versions, not sure if it would pick this up? But even if it did it probably wouldn't notice the version update needed in the `check_examples` script too.

Reply:
I don't think you can use `@latest` or something like that to specify the version of the pre-commit hooks. And, sadly, I think dependabot does not support [updating pre-commit hooks](https://github.com/dependabot/dependabot-core/issues/1524) for now. However, there exists a `pre-commit autoupdate` (https://pre-commit.com/#updating-hooks-automatically) command to update the hooks to the latest version. We could combine this with a GH action that creates a PR if a change is detected.

If we go this route, then I think it makes sense to have the black version in the `.pre-commit-config.yaml` be the single source of truth. We can then run `pre-commit run --all-files black` in the CI. That would mean, though, that installing `pre-commit` would be _kind of_ mandatory for the pytket repo. (There's also a way fo automatically enabling pre-commit on repositories: https://pre-commit.com/#automatically-enabling-pre-commit-on-repositories).

What do you think?

Reply:
Fine with me. Having a single source of truth is good enough I think.

Reply:
I've created #219 for taking care of the auto-update of black and other pre-commit hooks.